developer_guides/uuid/index.rst

```diff
@@ -0,0 +1,118 @@
+.. _uuid:
+
+UUID Usage in SOF
+#################
+
+We use universally unique identifier (UUID) for component type
+definition in SOF, and for the whole SOF stack, the UUID usage is
+following below rules:
```

Comment:
Thanks @keyonjie for the patch. One generic comment that bothers me a bit is that UUID is really part of the solution. From a developer point view, developer doesn't really have a problem that "I want to use UUID". Rather developers want to add components and use SOF infra to describe their component's interface. UUID is one aspect SOF provides to solve this. I think it's ok to have a separate section that summarizes the UUID implementation and semantics, but looking at the painpoints of our recent discussions w.r.t. component type and UUID, there are questions that go outside the UUID discussion.

Reply:
mark this resolved since this is addressed.